python/Ganga/GPIDev/Lib/Job/Job.py

```diff
@@ -2151,7 +2149,7 @@ def check_changability(obj1, obj2):
     def _commit(self, objects=None):
         """ Helper method to unconditionally commit to the repository. The 'objects' list specifies objects
         to be commited (for example the subjobs). If objects are not specified then just the self is commited """
-
+        # TODO This function should force a flush. However it is currently used everywhere which causes too many flushes.
         self._getRoot()._setDirty()
```

Comment:
Surely the calls to this should be wrapped behind something sensible.
I think _commit should only be being called when the status changes and this can easily be set to exclude all changes to which aren't ```'submitting'``` or ```'submitted'```?